NGitLab.Mock.Tests/GitLabClientMockTest.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using NGitLab.Mock.Config;
+using NGitLab.Models;
+using NUnit.Framework;
+
+namespace NGitLab.Mock.Tests;
+
+
+public class GitLabClientMockTest
+{
+    public static IEnumerable ProjectClientTestCases
+    {
+        get
+        {
+            static TestCaseData TestCase(string name, Func<IGitLabClient, ProjectId, object> getClient) => new TestCaseData(getClient).SetArgDisplayNames(name);
+
+            // Create a test case for each method in IGitLabClient that has a single parameter of type ProjectId
+            foreach (var method in GetMethods<ProjectId>())
+            {
+                yield return TestCase(method.Name, (client, id) => method.Invoke(client, new object[] { id }));
+            }
+        }
+    }
+
+    public static IEnumerable GroupClientTestCases
+    {
+        get
+        {
+            static TestCaseData TestCase(string name, Func<IGitLabClient, GroupId, object> getClient) => new TestCaseData(getClient).SetArgDisplayNames(name);
+
+            // Create a test case for each method in IGitLabClient that has a single parameter of type GroupId
+            foreach (var method in GetMethods<GroupId>())
+            {
+                // GetGroupMergeRequest() is not implemented in the mock MergeRequestClient and will throw a NotImplementedException
+                // To avoid a test failure, exclude this mehtod from the test cases (this is the only such case)
+                // This can be removed if a mock for group merge requests is implemented
+                if (string.Equals(method.Name, nameof(IGitLabClient.GetGroupMergeRequest), StringComparison.Ordinal))
```

Comment:
To avoid to miss tests if the `MergeRequestClient` ctor with GroupId is implemented at some point, can you also add a comment in the ctor above the throw? Stating that this exlusion should be removed.